### DISCUSSION TITLE
Usage with Ionic

### DISCUSSION THREAD

**Discussion by patriksh** [rocwang/vue-virtual-scroll-grid#209] [Q&A]
Hello!

I'm trying to use this with Ionic Vue and I got a bit stuck.
The first page of items loads but as I scroll nothing happens. I can reach the bottom and no more items are being loaded.

I added a log print in my pageProvider and it doesn't get called at all while scrolling, only when the component is first loaded.

Do you have any ideas on how could I debug this?

Thanks.

P.S. Another weird thing that's happening is that there's like 50px of empty space between each item.

**Comment by rocwang** [comment]
The component is a bit of sensitive on how the container and items are styled in CSS grid. So it might have something to do with that.

For example: https://github.com/rocwang/vue-virtual-scroll-grid#the-probe-slot

Ideally, could you provide a minimal repo that replicate the issue you have, which is easier for me to do the troubleshooting for you?

**Reply by patriksh** [reply]
Can you check it on Stackblitz?

https://stackblitz.com/edit/ionic-vue-virtual-scroll-grid?file=src%2Fcomponents%2FExample.vue
https://ionic-vue-virtual-scroll-grid--3000.local.webcontainer.io/

**Reply by rocwang** [reply]
It looks like the the component has trouble to detect the scroll events. It still can detect the window resize events and act accordingly.

Not sure if it has something to do to Ionic, which interferes with the scroll events.

Try chucking `window.addEventListener('scroll', console.log);` to your component and see if the event fires?

**Reply by patriksh** [reply]
Nope, nothing prints.
ion-content has it's own set of events it can fire. https://ionicframework.com/docs/api/content

**Reply by patriksh** [reply]
I did manage to use vue-virtual-scroller successfully with Ionic though, but I didn't look much into how it works.

**Reply by rocwang** [reply]
`vue-virtual-scroll-grid` use `window.addEventListener('scroll', ...)` [internally](https://github.com/rocwang/vue-virtual-scroll-grid/blob/main/src/utilites.ts#L42-L56).

Not sure it has to use Ionic custom scroll event in this case.
